### PR TITLE
getNamespaceByPrefix now checks topology for namespace existence

### PR DESCRIPTION
**Pull request by turetske**
-- Also the namespace struct contains a field determining whether the namespace is from topology

Addresses the first three points in #1005

### CODE REVIEW

**Review by haoming29 — DISMISSED** [review]
Some changes to consider

**Comment by haoming29** [comment]
Sorry I meant to hit `Changes Requested`

**Review by haoming29 — CHANGES_REQUESTED** [review]
Some changes to consider (see comments above)

File: registry/migrations/20240212192712_create_db_tables.sql [hunk 1]
```diff
@@ -6,7 +6,8 @@ CREATE TABLE IF NOT EXISTS namespace (
   pubkey TEXT NOT NULL,
   identity TEXT,
   admin_metadata TEXT CHECK (length("admin_metadata") <= 4000),
-  custom_fields TEXT CHECK (length("custom_fields") <= 4000) DEFAULT ''
+  custom_fields TEXT CHECK (length("custom_fields") <= 4000) DEFAULT '',
+  topology boolean
```
[the remark this answers is not in the document]
Comment (turetske) [reply registry/migrations/20240212192712_create_db_tables.sql:10]: Whenever I tried to add an extra Topology field in the namespace struct, the DB complained that the topology field didn't exist. Even if the namespace struct didn't contain it, the db was expecting it.

File: registry/registry_db.go
[the remark this answers is not in the document]
Comment (turetske) [reply registry/registry_db.go:1]: The topology table doesn't have a public key

**Review by haoming29 — APPROVED** [review]
LGTM! Thanks for fixing this

**Pull request merged**